### PR TITLE
chore: update types.ts for all generator to document all possible options

### PR DESCRIPTION
**Pull request by nmerget**
## Description

Please provide the following information:

Align all generators to have an `index.ts`, `generator.ts` and `types.ts` files.
This will improve:
- the structure of the repo
- searchability of options and metadata

Make sure to follow the PR preparation steps in [CONTRIBUTING.md](../CONTRIBUTING.md#preparing-your-pr) before submitting your PR:

- [x] format the codebase: from the root, run `yarn fmt:prettier`.
- [x] update all snapshots (in core & CLI): from the root, run `yarn test:update`
- [x] add Changeset entry: from the root, run `yarn g:changeset` and follow the CLI instructions. Alternatively, use the Changeset Github Bot to create the file.

### CODE REVIEW

File: packages/core/src/types/metadata.ts [hunk 1]
```diff
@@ -11,27 +26,26 @@ export type ComponentMetadata = {
...
-  };
-  solid?: {
-    state?: Dictionary<SolidState>;
-  };
-  rsc?: {
-    componentType?: 'client' | 'server';
-  };
+  tagName?: string; // Deprecated: Use this for web-components to change the tagName
+  forwardRef?: string; // Deprecated: Use this for react forwardRef
+  isAttachedToShadowDom?: boolean; // Enables shadowDom for web-components
```
Comment (samijaber) [inline packages/core/src/types/metadata.ts:31]: if you use JSDoc @deprecated syntax, the comments will flow to the end-user when using `useMetadata`:

```suggestion
  /** @deprecated Use this for web-components to change the tagName  */
  tagName?: string;
  /** @deprecated Use this for react forwardRef */
  forwardRef?: string;
  /** Enables shadowDom for web-components */
  isAttachedToShadowDom?: boolean;
```

**Review by samijaber — COMMENTED** [review]
Thanks for tidying things up!

**Comment by samijaber** [comment]
You missed adding the changeset for this PR. :)

**Comment by nmerget** [comment]
Sometimes a ci step fails with:

`YN0009: nx@npm:19.0.8 [d6dc7] couldn't be built successfully (exit code 129, logs can be found here: /tmp/xfs-b1b31526/build.log)`

There is no additional error log, do you have any clue about it?

**Comment by samijaber** [comment]
> Sometimes a ci step fails with:
> 
> `YN0009: nx@npm:19.0.8 [d6dc7] couldn't be built successfully (exit code 129, logs can be found here: /tmp/xfs-b1b31526/build.log)`
> 
> There is no additional error log, do you have any clue about it?

This is an odd issue impacting Nx in multiple Github CI servers of ours. I am not really sure what's going on...

**Comment by samijaber** [comment]
this is approved, but there are some merge conflicts to resolve first

**Comment by nmerget** [comment]
> this is approved, but there are some merge conflicts to resolve first

Solved them :)

**Review by samijaber — APPROVED** [review]

**Pull request merged**